systest/tests/common.go

```diff
@@ -13,6 +13,7 @@ import (
 	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap"
+	_ "go4.org/unsafe/assume-no-moving-gc"
```

Comment:
Could you explain why is it needed?

Reply:
https://github.com/spacemeshos/go-spacemesh/actions/runs/6075215589/job/16480953573

Reply:
It still fails though, I'll look into it tomorrow again.

Reply:
I cannot figure out which of our dependencies uses it. Do you know?

Reply:
`github.com/francoispqt/gojay@v1.2.13` -> `cloud.google.com/go@v0.37.0` -> `golang.org/x/build@v0.0.0-20190111050920-041ab4dc3f9d `-> `go4.org@v0.0.0-20180809161055-417644f6feb5`

`github.com/francoispqt/gojay@v1.2.13` is used by `go-libp2p` and some of its dependencies (like `ipfs` and `quic-go`).

Unfortunately it seems like the library has been abandoned around 2019. It uses some very outdated packages (like the ones listed above) and has known security issues 😞, but the author hasn't responded to any PRs or Issues in a few years.

We can probably "fix" it by setting the environment variable that skips the check at the risk that if we update to 1.22 or any later version of Go and the GC behavior changes we see crashes.

Reply:
My reasoning for this is:

- In Go 1.21 the check still passes: see (https://github.com/go4org/unsafe-assume-no-moving-gc/blob/e7c30c78aeb2bf1ab602f2e789b86db8d9942388/check.go#L19-L25 and https://github.com/golang/go/blob/c19c4c566c63818dfd059b352e52c4710eecf14d/src/runtime/mgc.go#L170-L172)
- If go would resolve the version I specified in `go.mod` the check would pass, but for some reason it uses `go4.org/unsafe/assume-no-moving-gc@v0.0.0-20230204201903-c31fa085b70e` instead of `go4.org/unsafe/assume-no-moving-gc v0.0.0-20230525183740-e7c30c78aeb2` when performing the check, which is an older version that didn't have the check for Go 1.21 yet.

Reply:
Using version `v4.0.0-pre1` solves the issue 🙂

Reply:
We don't use `go4.org/unsafe/assume-no-moving-gc` in go-spacemesh, that was an incorrect assumption of mine, but `gomplate` does. The error wasn't from go-spacemesh, but from `gomplate`. Using a newer version solves the issue 🙂